internal/pkg/cli/pipeline_override.go

```diff
@@ -0,0 +1,157 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ssm"
+	"github.com/aws/copilot-cli/internal/pkg/aws/identity"
+	"github.com/aws/copilot-cli/internal/pkg/aws/sessions"
+	"github.com/aws/copilot-cli/internal/pkg/config"
+	"github.com/aws/copilot-cli/internal/pkg/term/log"
+	termprogress "github.com/aws/copilot-cli/internal/pkg/term/progress"
+	"github.com/aws/copilot-cli/internal/pkg/term/prompt"
+	"github.com/aws/copilot-cli/internal/pkg/term/selector"
+	"github.com/aws/copilot-cli/internal/pkg/workspace"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+type overridePipelineOpts struct {
+	*overrideOpts
+
+	// Interfaces to interact with dependencies.
+	ws       wsPipelineReader
+	wsPrompt wsPipelineSelector
+}
+
+func newOverridePipelineOpts(vars overrideVars) (*overridePipelineOpts, error) {
+	fs := afero.NewOsFs()
+	ws, err := workspace.Use(fs)
+	if err != nil {
+		return nil, err
+	}
+
+	sessProvider := sessions.ImmutableProvider(sessions.UserAgentExtras("pipeline override"))
+	defaultSess, err := sessProvider.Default()
+	if err != nil {
+		return nil, fmt.Errorf("default session: %v", err)
+	}
+
+	cfgStore := config.NewSSMStore(identity.New(defaultSess), ssm.New(defaultSess), aws.StringValue(defaultSess.Config.Region))
+
+	prompt := prompt.New()
+
+	cmd := &overridePipelineOpts{
+		overrideOpts: &overrideOpts{
+			overrideVars: vars,
+			fs:           fs,
+			cfgStore:     cfgStore,
+			prompt:       prompt,
+			cfnPrompt:    selector.NewCFNSelector(prompt),
+			spinner:      termprogress.NewSpinner(log.DiagnosticWriter),
+		},
+		ws:       ws,
+		wsPrompt: selector.NewWsPipelineSelector(prompt, ws),
+	}
+	cmd.overrideOpts.packageCmd = cmd.newPipelinePackageCmd
+	return cmd, nil
+}
+
+// Validate returns an error for any invalid optional flags.
+func (o *overridePipelineOpts) Validate() error {
+	if err := o.overrideOpts.Validate(); err != nil {
+		return err
+	}
+	return o.validatePipelineName()
+}
+
+// Ask prompts for and validates any required flags.
+func (o *overridePipelineOpts) Ask() error {
+	if o.name == "" {
+		if err := o.askPipelineName(); err != nil {
+			return err
+		}
+	}
+	return o.overrideOpts.Ask()
+}
+
+// Execute writes IaC override files to the local workspace.
+func (o *overridePipelineOpts) Execute() error {
+	o.overrideOpts.dir = func() string {
+		return o.ws.PipelineOverridesPath(o.name)
+	}
+	return o.overrideOpts.Execute()
+
+}
+
+func (o *overridePipelineOpts) validatePipelineName() error {
+	if o.name == "" {
+		return nil
+	}
+	pipeline_list, err := o.ws.ListPipelines()
```

Comment:
```suggestion
	pipelines, err := o.ws.ListPipelines()
```
https://google.github.io/styleguide/go/decisions#repetitive-with-type

Reply:
I don't know sometimes why I get the file name styling to these variables🤦‍♂️